lib/serve/functions.js

```diff
@@ -75,44 +83,62 @@ function _startEmulator(options) {
     utils.logWarning(chalk.yellow('functions:') + ' The emulator works best with Node version v6.9.1, you have ' + process.version + '\n');
   }
   var functionsDir = path.join(options.config.projectDir, options.config.get('functions.source'));
-  var projectId = getProjectId(options);
-  var controllerConfig = _.merge({tail: true, projectId: projectId}, _getPorts(options));
+  var controllerConfig = _.merge({tail: true}, _getPorts(options));
   controller = new EmulatorController(controllerConfig);
+  var firebaseConfig;
 
-  return controller.start({ detached: false, stdio: ['ignore', process.stdout, process.stderr]}).then(function() {
+  return functionsConfig.getFirebaseConfig(projectId, options.instance)
+  .then(function(result) {
+    firebaseConfig = JSON.stringify(result);
+    process.env.FIREBASE_PROJECT = firebaseConfig;
+    process.env.DB_NAMESPACE = options.instance;
```

Comment:
Good point, DB_NAMESPACE used to be needed by the SDK during trigger parsing, it is no longer needed. I've removed this from both the serve and deploy process.